Model/ExtendedCreditMemoManagement.php

```diff
@@ -0,0 +1,55 @@
+<?php
+namespace SnowIO\ExtendedSalesRepositories\Model;
+
+use Magento\Sales\Api\Data\CreditmemoInterface;
+
+/**
+ * Class ExtendedCreditMemoManagement
+ *
+ * @package SnowIO\ExtendedSalesRepositories\Model
+ */
+class ExtendedCreditMemoManagement
+{
+    /**
+     * Never return the shipping amount
+     * @param CreditmemoInterface $creditmemo
+     * @author Alexander Wanyoike <aw@amp.co>
+     */
+    public static function resetAmounts(CreditmemoInterface $creditmemo)
+    {
+        $creditmemo
+            ->setBaseShippingAmount(0)
+            ->setShippingAmount(0)
+            ->setBaseSubtotal(0)
+            ->setBaseGrandTotal(0)
+            ->setGrandTotal(0);
+    }
+
+    /**
+     * Reset tax calculations
+     * @param CreditmemoInterface $creditmemo
+     * @author Alexander Wanyoike <aw@amp.co>
+     */
+    public static function resetTax(CreditmemoInterface $creditmemo)
+    {
+        $creditmemo->setBaseTaxAmount(0)
```

Comment:
Again, this was done for Seraphine specific functionality because the client did not want to refund the shipping amount (maybe we don't want to by default I guess?) which involved resetting all tax and shipping calculations then calling `calculateTotals`

Reply:
I noticed that if we didn't include the changes that were made in seraphine the totals calculation are inconsistent. I thing Ill remove the shipping amount changes cause there might be some cases that shipping amount might be taken into account again. I reckon this will need a further discussion with andrew.